common-api/src/main/java/org/pmiops/workbench/cdr/dao/ConceptService.java

```diff
@@ -74,47 +79,61 @@ public static String modifyMultipleMatchKeyword(String query){
                     } catch (NumberFormatException e) {
                         // Not a long, don't try to match it to a concept ID.
                     }
-                    predicates.add(criteriaBuilder.or(queryPredicates.toArray(new Predicate[0])));
+
+                    List<Predicate> conceptName_Filter = new ArrayList<>();
+
+
+                    Expression<Double> matchExp = criteriaBuilder.function("match", Double.class,
+                            root.get("conceptName"), criteriaBuilder.parameter(String.class , "keyword"));
+                    conceptName_Filter.add(criteriaBuilder.greaterThan(matchExp, 0.0));
+
 
                     // Optionally filter on standard concept, vocabulary ID, domain ID
                     if (standardConceptFilter.equals(StandardConceptFilter.STANDARD_CONCEPTS)) {
-                        predicates.add(criteriaBuilder.equal(root.get("standardConcept"),
+                        conceptName_Filter.add(criteriaBuilder.equal(root.get("standardConcept"),
                                 criteriaBuilder.literal(STANDARD_CONCEPT_CODE)));
                     } else if (standardConceptFilter.equals(StandardConceptFilter.NON_STANDARD_CONCEPTS)) {
                         List<Predicate> standardConceptPredicates = new ArrayList<>();
                         standardConceptPredicates.add(criteriaBuilder.isNull(root.get("standardConcept")));
                         standardConceptPredicates.add(criteriaBuilder.notEqual(root.get("standardConcept"),
                                 criteriaBuilder.literal(STANDARD_CONCEPT_CODE)));
-                        predicates.add(criteriaBuilder.or(
+                        conceptName_Filter.add(criteriaBuilder.or(
                                 standardConceptPredicates.toArray(new Predicate[0])));
                     }
+
+                    predicates.add(
+                            criteriaBuilder.or(
+                                    criteriaBuilder.or(conceptCode_Id.toArray(new Predicate[0])),
+                                    criteriaBuilder.and(conceptName_Filter.toArray(new Predicate[0]))
+                            ));
+
                     if (vocabularyIds != null) {
                         predicates.add(root.get("vocabularyId").in(vocabularyIds));
                     }
                     if (domainIds != null) {
                         predicates.add(root.get("domainId").in(domainIds));
                     }
+
                     return criteriaBuilder.and(predicates.toArray(new Predicate[0]));
                 };
-
         // Return up to limit results, sorted in descending count value order.
         Pageable pageable = new PageRequest(0, limit,
                 new Sort(Direction.DESC, "countValue"));
         NoCountFindAllDao<Concept, Long> conceptDao = new NoCountFindAllDao<>(Concept.class,
                 entityManager);
+        conceptDao.keyword=keyword;
```

Comment:
If we need to do this, let's pass it in to the constructor, and make the field final, instead of assigning it like this.